api_app/analyzers_manager/observable_analyzers/circl_pdns.py

```diff
@@ -8,21 +8,17 @@
 
 from api_app.exceptions import AnalyzerRunException
 from api_app.analyzers_manager import classes
-from intel_owl import secrets
 
 
 class CIRCL_PDNS(classes.ObservableAnalyzer):
-    def set_config(self, _):
-        self.__credentials = secrets.get_secret("CIRCL_CREDENTIALS")
+    def set_params(self, params):
+        self.__credentials = self._secrets["pdns_credentials"]
         self.domain = self.observable_name
         if self.observable_classification == "url":
```

Comment:
`if self.observable_classification == "url":`

instead of hardcoding `url`, use `AnalyzerConfigSerializer.ObservableTypes.URL`